File: Source/CSharp/Expressive/Expressive/Operators/Logic/NotOperator.cs
using Expressive.Expressions;
using System.Collections.Generic;
using System.Linq;

namespace Expressive.Operators.Logic
{
    internal class NotOperator : OperatorBase
    {
        #region IOperator Members

        public override string[] Tags { get { return new[] { "!", "not" }; } }

        public override IExpression BuildExpression(Token previousToken, IExpression[] expressions, ExpressiveOptions options)
        {
            return new UnaryExpression(UnaryExpressionType.Not, expressions[0] ?? expressions[1]);
        }

        public override OperatorPrecedence GetPrecedence(Token previousToken)
        {
            return OperatorPrecedence.Not;
        }

        #endregion
    }
}

File: Source/Java/Expressive/src/main/java/com/bijington/expressive/operators/logic/NotOperator.java
package com.bijington.expressive.operators.logic;

import com.bijington.expressive.expressions.*;
import com.bijington.expressive.operators.OperatorBase;
import com.bijington.expressive.operators.OperatorPrecedence;

/**
 * Created by shaun on 11/07/2016.
 */
public class NotOperator extends OperatorBase {
    @Override
    public String[] getTags() { return new String[] { "!", "not" }; }

    @Override
    public IExpression buildExpression(String previousToken, IExpression[] expressions) {
        return new UnaryExpression(UnaryExpressionType.Not, (expressions[0] != null) ? expressions[0] : expressions[1]);
    }

    @Override
    public int getPrecedence(String previousToken) {
        return OperatorPrecedence.Not;
    }
}